openstack/blockstorage/v2/noauth/urls.go

```diff
@@ -0,0 +1,17 @@
+package noauth
+
+import (
+	"fmt"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+func getURL(client *gophercloud.ProviderClient, eo EndpointOpts, clientType string) (string, error) {
+	url := ""
+	if len(eo.CinderEndpoint) > 0 && clientType == "volumev2" {
```

Comment:
How about moving this to `initClientOpts`? That way, when `getURL` is called, we can be confident that data is valid and just return a string.